### PR TITLE
Move ImageProcessing to KA, add DeviceSideRNG and FunWithArrays

### PR DESCRIPTION
**Pull request by jpsamaroo**
@vchuravy I'm having trouble getting the kernels to compile; GPUCompiler complains about calls to `jl_f_tuple`, but not for all of the KA kernels.

### CODE REVIEW

**Comment by vchuravy** [comment]
Can we use Jupyter notebooks instead? Pluto's execution model makes OOM with GPUs more likely

(I know I drafted that first Pluto notebook)

**Comment by jpsamaroo** [comment]
Yeah, this ended up messing with the GPU far less :smile:

File: common_definitions.jl [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+using KernelAbstractions
+
+if GPU_PKG_NAME == "CUDA"
+using CUDA, CUDAKernels
+const GPUMOD = CUDA
+const GpuArray = CuArray
+const GpuBackend = CuDevice()
+elseif GPU_PKG_NAME == "AMDGPU"
+using AMDGPU, ROCKernels
```
Comment (maleadt) [inline common_definitions.jl]: I'd rather not import multiple packages. It's unclear how CUDA.jl's overrides interact with AMDGPU/oneAPI (probably badly).

Comment (maleadt) [reply common_definitions.jl]: Ah never mind, it's conditional.

**Pull request merged**